spock-core/core.gradle

```diff
@@ -13,6 +13,7 @@ Spock is inspired from JUnit, jMock, RSpec, Groovy, Scala, Vulcans, and other fa
 dependencies {
   compile libs.groovy // easiest way to add Groovy dependency to POM
   compile libs.junit
+  compile libs.annotations
```

Comment:
Java ignores missing annotations so there shouldn't be any issues with 
setting this dependency to optional. If someone has those annotations in his classpath it will they will be used, otherwise they'll be dropped.
https://stackoverflow.com/questions/3567413/why-doesnt-a-missing-annotation-cause-a-classnotfoundexception-at-runtime